packages/react-native/src/theme/utils.ts

```diff
@@ -0,0 +1,26 @@
+import mapValues from 'lodash/mapValues';
+import get from 'lodash/get';
+import { Tokens } from './types';
+
+/**
+ * Util similar to lodash mapValues but recurses into nested objects.
+ * @param obj
+ * @param callback
+ */
+const mapValuesDeep = <T>(obj: T, callback: Function): T =>
+  typeof obj === 'object'
+    ? (mapValues(obj, (obj) => mapValuesDeep(obj, callback)) as T) // cast back to T to appease TS
+    : (callback(obj) as T);
+
+/**
+ * Function that will walk down the token object
+ * and perform the mapValuesDeep function on each token.
+ */
+export const setupTokens = (tokens: Tokens): Tokens =>
```

Comment:
Think maybe we have a disconnect on abstracting this callback in to it's own function. What I have been trying to express is that the below should be its own util:
```
const callback = (value: string) => {
    return typeof value === 'string' &&
      value.startsWith('{') &&
      value.endsWith('}')
      ? (get(tokens, value.substring(1, value.length - 2)) as string) // cast back to string to appease TS, as lodash get returns any
      : value;
  }
```

And still think we need to add explicit notes around the string parsing behavior, in particular due to how fragile it is

Reply:
Are you suggesting we should pass in `tokens` to the callback fn and also in `mapValuesDeep`?

Reply:
Why are rewriting the string parsing behavior when we already have it in the UI package (which it gets from style dictionary)? That parsing/resolving behavior is thoroughly tested and used by lots of packages. Not a blocker to get started, but I would feel nervous about having different functions for React Native and React Web.